lab_4_summarization_textrank/README.md

```diff
@@ -0,0 +1,758 @@
+2. После этого вес каждой вершины пересчитывается по этой формуле:
+$$S(V_{i}^{k}) = (1 - d) + d \cdot \sum_{j \in InOut(V_{i})} \frac{1}{|InOut(V_{j})| } S(V_{j}^{k-1})$$
+
+    Обозначения:
+    * $V$ - вершина графа (то есть слово)
+    * $d$ - damping factor, то есть вероятность перейти от этой вершины к любой другой, его значение обычно равно 0.85
+    * $InOut(V)$ - вершины, инцидентные вершине $V$
+    * $|InOut(V)|$ - количество вершин, инцидентных с $V$ (то есть, как мы назвали это в предыдущем пункте, `InOut score`)
+    * $k$ - это номер итерации: при переподсчёте веса вершины мы опираемся на веса инцидентных с ней вершин с *прошлых итераций*
+    
+    Таким образом, чтобы посчитать новый вес для новой вершины, нужно:
+    * перебрать все вершины, инцидентные с данной вершиной
+    * для каждой из таких вершин найти ее `InOut score`, использовать его как знаменатель в дроби $\frac{1}{|InOut(V_{j})|}$ и умножить полученное значение на вес этой же вершины
+    * полученные значения просуммировать и умножить на $d$
+    * далее результат умножения сложить с $(1 - d)$, и результат такого сложения становится новым весом рассматриваемой вершины
+
+3. После такой оценки важности необходимо узнать, насколько сильно веса вершин изменились: для этого из веса каждой вершины вычитается ее предыдущий вес. Полученные разности суммируются и сравниваются с *порогом сходимости* - это какое-то маленькое заранее заданное число. 
+В нашем случае оно равно `0.0001`. Если сумма разностей весов оказалась меньше, чем это маленькое число, то можно сказать, что вес вершин почти не изменился. Это значит, что оптимальные веса были найдены, алгоритм завершается. В противном случае мы продолжаем переоценку важности вершин, возвращаясь в пункт 2.
+
+Данный алгоритм оперировал словами и находил ключевые слова для текста. Чтобы позволить этому алгоритму находить наиболее важные предложения, 
+необходимо внести несколько модификаций (модификации выделены жирным шрифтом): 
+
+* $V$ - вершина графа (то есть **предложение**)
+* $d$ - damping factor, то есть вероятность перейти от этой вершины к любой другой, его значение обычно равно `0.85`
+* $InOut(V)$ - вершины, **похожие на вершину** $V$
+* $|InOut(V)|$ - количество вершин, **похожих на** $V$ (то есть, как мы назвали это в предыдущем пункте, `InOut score`)
+* $k$ - это номер итерации: при переподсчёте веса вершины мы опираемся на веса инцидентных с ней вершин с *прошлых итераций*
+    
+Таким образом, чтобы посчитать новый вес для новой вершины, нужно:
+* перебрать все вершины, инцидентные с данной вершиной
+* для каждой из таких вершин найти ее `InOut score`, использовать его как знаменатель в дроби $\frac{1}{|InOut(V_{j})|}$ и умножить полученное значение на вес этой же вершины
+* полученные значения просуммировать и умножить на $d$
+* далее результат умножения сложить с $(1 - d)$; результат такого сложения становится новым весом рассматриваемой вершины
+
+В этом шаге Вы будете реализовывать `TextRankSummarizer`.
```

Comment:
a lot of text, but, in reality, you could just say that learning algorithm is the same, but what changes is meaning of elements participating in the key formula. otherwise it seems like a completely different algorithm to implement